python/tskit/trees.py

```diff
@@ -2970,27 +2992,29 @@ def variants(self, as_bytes=False, samples=None, impute_missing_data=False):
             The ``as_bytes`` parameter is kept as a compatibility
             option for older code. It is not the recommended way of
             accessing variant data, and will be deprecated in a later
-            release. Another method will be provided to obtain the allelic
-            states for each site directly.
+            release.
 
         :param bool as_bytes: If True, the genotype values will be returned
-            as a Python bytes object. This is useful in certain situations
-            (i.e., directly printing the genotypes) or when numpy is
-            not available. Otherwise, genotypes are returned as a numpy
-            array (the default).
+            as a Python bytes object. Legacy use only.
         :param array_like samples: An array of sample IDs for which to generate
             genotypes, or None for all samples. Default: None.
         :param bool impute_missing_data: If True, the allele assigned to any
             isolated samples is the ancestral state; that is, we impute
             missing data as the ancestral state. Default: False.
+        :param tuple alleles: A tuple of strings describing the encoding of
+            alleles to genotype values. At least one allele must be provided.
+            If duplicate alleles are provided, output genotypes will always be
+            encoded as the first occurance of the allele. If None (the default),
+            the alleles are encoded as they are encountered during genotype
+            generation.
```

Comment:
maybe say "describing how genotypes are encoded as integers" (since "genotype values" is ambiguous)

Reply:
or maybe "describing how genotypes are encoded as integers, when integers are used (e.g., in :meth:.genotype_matrix())"